packages/electron-filestore/filestore.js

```diff
@@ -82,24 +84,37 @@ class FileStore {
    * Loads persisted device info. If none is present, it generates an identifier
    * for the device and persists it prior to returning device info
    */
-  async getDeviceInfo () {
-    try {
-      const contents = await readFile(this._paths.device)
-      const device = JSON.parse(contents)
-      if (!device.id) {
-        device.id = createIdentifier()
-        await this.setDeviceInfo(device)
+  getDeviceInfo () {
+    return new Promise((resolve, reject) => {
+      const ensureDeviceId = async (device = {}) => {
+        if (!device.id) {
+          device.id = createIdentifier()
+        }
+        // this intentionally resolves the device field before async'ly persisting device info
+        // – it means the getDeviceInfo() function returns device info as soon as it's available,
+        // rather than having to wait until it's been persisted to disk
+        resolve(device)
+        try {
+          await this.setDeviceInfo(device)
+        } catch (e) {
+          // it's too late to do anything with this error, so just swallow it
+        }
```

Comment:
Having this logic after the `resolve()` means that the `{ id }` value can be returned/resolved asap. However, if the device repeatedly fails to write to this storage, a different ID will be generated every app run. Is this an acceptable tradeoff?

Reply:
It makes sense. If it can't be written immediately, it might never be written ever, though that computer will always send fully formed events/sessions. There's also nothing we can do about it if it fails to write in any case.